src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -395,15 +439,14 @@ private async Task DownloadToOfflineCacheAsync(IEnumerable<WorkloadId> workloadI
             }
         }
 
-        private IEnumerable<PackInfo> GetPacksToInstall(IEnumerable<WorkloadId> workloadIds)
+        IEnumerable<PackInfo?> GetPacksToInstall(IEnumerable<WorkloadId> workloadIds)
```

Comment:
If you ever figure out how to make Roslyn tell this should be non-nullable becase you do `Where(not null)`, let me know.

I know about `Select(x => x!)` and `Cast<>()` but they seem like extra leg work..

Reply:
I know how to, you can use the more SQL looking syntax, I used it here: https://github.com/dotnet/sdk/pull/24028/files#diff-498ad5982ef0f101943528170b9fb47715442c897bf94ef0c8c1f2438f4f8effR361

Not ideal, that's why I do it in the caller and not the callee.